src/components/tabs/tabs.spec.js

```diff
@@ -931,4 +933,171 @@ describe("Tabs", () => {
       });
     });
   });
+
+  describe("when children of Tab have validation failures", () => {
+    const MockComponent = ({ show = true, error, warning, info }) => (
+      <Tabs data-element="bar" data-role="baz">
+        <Tab
+          tabId="1"
+          title="Test"
+          errorMessage=""
+          warningMessage=""
+          infoMessage=""
+        >
+          {show && (
+            <Textbox
+              value="foo"
+              onChange={() => {}}
+              error={error}
+              warning={warning}
+              info={info}
+            />
+          )}
+        </Tab>
+      </Tabs>
+    );
+
+    it("any error in a child component is correctly reported in the TabTitle", () => {
+      const wrapper = mount(<MockComponent error />);
+
+      expect(wrapper.find(ValidationIcon).exists()).toBe(true);
+
+      wrapper.setProps({ show: false });
+      wrapper.update();
+      expect(wrapper.update().find(ValidationIcon).exists()).toBe(false);
+    });
+
+    it("any warning in a child component is correctly reported in the TabTitle", () => {
+      const wrapper = mount(<MockComponent warning />);
+
+      expect(wrapper.find(ValidationIcon).exists()).toBe(true);
+
+      wrapper.setProps({ show: false });
+      wrapper.update();
+      expect(wrapper.update().find(ValidationIcon).exists()).toBe(false);
+    });
+
+    it("any info in a child component is correctly reported in the TabTitle", () => {
+      const wrapper = mount(<MockComponent info />);
+
+      expect(wrapper.find(ValidationIcon).exists()).toBe(true);
+
+      wrapper.setProps({ show: false });
+      wrapper.update();
+      expect(wrapper.update().find(ValidationIcon).exists()).toBe(false);
+    });
+  });
```

Comment:
Minor - Could also be refactored into one it.each()

Reply:
Yeah good point, will update if I get any further comments